src/oas/oas-to-ms3/oas-30-to-ms3/index.ts

```diff
@@ -74,18 +81,100 @@ class MS3toOAS30toMS3 {
          contentType: <MS3Interface.contentType>key
       };
 
-      if (value.schema.$ref) {
+      if (value.schema && value.schema.$ref) {
         const splitArr: string[] = value.schema.$ref.split('/');
         const name: string = splitArr.pop();
-        convertedBody.type = name;
+        convertedBody.type = this.getRefId(name, 'schemas');
+      } else if (value.schema) {
+        convertedBody.type = v4();
+        value.schema.__id = convertedBody.type;
+        this.ms3API.dataTypes.push(value.schema);
       }
-      else {
-        // push schemas to datatypes
-        convertedBody.type = value.schema.name;
+
+      if (value.examples && value.examples.$ref) {
+        const splitArr: string[] = value.examples.$ref.split('/');
+        const name: string = splitArr.pop();
+        convertedBody.selectedExamples.push(this.getRefId(name, 'examples'));
+      } else if (value.examples) {
+        convertedBody.selectedExamples = this.convertExamples(<OAS30Interface.Example>value.examples);
       }
 
       resultArray.push(convertedBody);
+      return resultArray;
+    }, []);
+  }
+
+  getRefId(name: string, entity: 'schemas' | 'examples') {
```

Comment:
/**
 * Get by $ref new or existing id of the included entity in resulting Ms3 API
**/